dashboard/public/pixelation/pixelation.js

```diff
@@ -80,7 +80,14 @@ function initProjects() {
         options.projectData = levelSource;
       },
       getLevelSource: function () {
-        return pixel_data.value.replace(/[ \n]/g, "");
+        // Store the source in whichever format the level specifies.
+        if (isHexSelected()) {
+          var hexCode = pixel_data.value.replace(/[^0-9A-F]/gi, "");
+          return isHexLevel() ? hexCode : hexToBinPvt(hexCode);
+        } else {
+          var binCode = pixel_data.value.replace(/[^01]/gi, "");
+          return isHexLevel() ? binToHexPvt(binCode) : binCode;
+        }
```

Comment:
It looks like these regex's are to strip out invalid data. How do we get that invalid data? (I thought we already prevented you from entering it).